Lib/Src/CUartCom.cpp

```diff
@@ -98,8 +106,67 @@ void CUartCom::sendMessage(const std::string &msg)
     HAL_UART_Transmit(mp_huart, (uint8_t *)c_msg, msg_len, UART_TIMEOUT);
```

Comment:
This command is blocking. It will take as long to execute as it takes to actually transmit the message. It's a waste of resources. Have a go at using DMA transmission for this purposes. However, using DMA opens a whole bunch of other problems like concurrent access, so passing by reference without copying the messages is not going to work. ;-)

Reply:
Let's discuss if you are not clear about why this is a problem.